### PR TITLE
Add string interpolators for error logging

### PR DESCRIPTION
**Pull request by ioaoue**
As of SLF4J version 1.6.0, it is possible to log an exception along with a parameterized message. This PR adds some interpolators to use that feature.

### CODE REVIEW

File: logging/structured/src/main/scala/tofu/logging/impl/ContextLoggingImpl.scala [hunk 1]
```diff
@@ -33,14 +33,14 @@ class ContextLoggingImpl[F[_]: Applicative, C: Loggable, Service: ClassTag](cont
...
+  override def traceCause(message: String, cause: Throwable, values: LoggedValue*): F[Unit] =
+    context.ask(ctx => logger.trace(ContextMarker(ctx), message, values :+ cause: _*)).whenA(errorEnabled)
+  override def debugCause(message: String, cause: Throwable, values: LoggedValue*): F[Unit] =
+    context.ask(ctx => logger.debug(ContextMarker(ctx), message, values :+ cause: _*)).whenA(errorEnabled)
+  override def infoCause(message: String, cause: Throwable, values: LoggedValue*): F[Unit] =
+    context.ask(ctx => logger.info(ContextMarker(ctx), message, values :+ cause: _*)).whenA(errorEnabled)
+  override def warnCause(message: String, cause: Throwable, values: LoggedValue*): F[Unit] =
+    context.ask(ctx => logger.error(ContextMarker(ctx), message, values :+ cause: _*)).whenA(errorEnabled)
+  override def errorCause(message: String, cause: Throwable, values: LoggedValue*): F[Unit] =
+    context.ask(ctx => logger.error(ContextMarker(ctx), message, values :+ cause: _*)).whenA(errorEnabled)
```
Comment (ioaoue) [inline logging/structured/src/main/scala/tofu/logging/impl/ContextLoggingImpl.scala]: I don't know how this change will affect performance

File: logging/structured/src/main/scala/tofu/logging/Logging.scala [hunk 2]
```diff
@@ -33,8 +33,8 @@ trait LoggingBase[F[_]] {
     write(level, message, values: _*)
 
   /** could be overridden in the implementations, write message about some exception */
-  def writeCause(level: Level, message: String, cause: Throwable): F[Unit] =
-    write(level, message, Logging.errorvalue(cause))
+  @silent def writeCause(level: Level, message: String, cause: Throwable, values: LoggedValue*): F[Unit] =
+    write(level, message, values: _*)
```
Comment (ioaoue) [inline logging/structured/src/main/scala/tofu/logging/Logging.scala]: I think the previous version could do something strange. So I changed it a bit.

File: logging/structured/src/main/scala/tofu/logging/impl/ContextLoggingImpl.scala (hunk 1 above)
Comment (Odomontois) [reply logging/structured/src/main/scala/tofu/logging/impl/ContextLoggingImpl.scala]: I don't understand the reason why we need to append `cause` to values here. It seems counter-intuitive

Comment (Odomontois) [reply logging/structured/src/main/scala/tofu/logging/impl/ContextLoggingImpl.scala]: We'll have the `cause` in the log message available to a layout anyway.

File: logging/structured/src/main/scala/tofu/logging/Logging.scala (hunk 2 above)
Comment (Odomontois) [reply logging/structured/src/main/scala/tofu/logging/Logging.scala]: Now we lost the `cause`, I suppose appending `values :+ Logging.errorvalue(cause)` would be more suitable here

File: logging/structured/src/main/scala/tofu/logging/impl/ContextLoggingImpl.scala (hunk 1 above)
Comment (Odomontois) [reply logging/structured/src/main/scala/tofu/logging/impl/ContextLoggingImpl.scala]: Oh. I get it. logger doesn't have API for messages with interpolation as well as cause.
I suppose we must not throw cause argument here and break compatibility with existing formats handling causes.
Instead, you can do some wrapping values in the Marker

File: logging/structured/src/main/scala/tofu/logging/Logging.scala (hunk 2 above)
Comment (Odomontois) [reply logging/structured/src/main/scala/tofu/logging/Logging.scala]: I propose next changes:

1. Move `Logging.errorvalue` to `LoggedValue.error`
2. Change the result of `LoggedValue.error` to the new wrapper
   that is sublclass of both `LoggedValue` and `Throwable(cause.message, cause)`.

**Pull request merged**